test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -6252,7 +6253,33 @@ def submit(self, obj, script=None, extend=None, submit_dir=None):
         rc = None
 
         if isinstance(obj, Job):
-            if script is None and obj.script is not None:
+            if self.platform == 'cray' or self.platform == 'craysim':
+                m = None
+                vncompute = False
+                if 'Resource_List.select' in obj.attributes:
+                    m = re.match('.*vntype=cray_compute.*',
+                                 obj.attributes['Resource_List.select'])
+                if 'Resource_List.vntype' in obj.attributes:
+                    vn_type = obj.attributes['Resource_List.vntype']
+                    if vn_type == 'cray_compute':
+                        vncompute = True
+                if obj.script is not None:
+                    script = obj.script
+                elif m is not None or vncompute:
+                    aprun_cmd = "aprun -B"
+                    if script:
+                        aprun_cmd += " " + script
+                    else:
+                        aprun_cmd += " " + obj.attributes[ATTR_executable]
+                        arg_list = obj.attributes[ATTR_Arglist]
+                        aprun_cmd += " " + self.utils.convert_arglist(arg_list)
+                    (fd, fn) = self.du.mkstemp(hostname=None,
+                                               prefix='PtlPbsJobScript',
+                                               uid=None, gid=None, mode=0755,
+                                               body=aprun_cmd)
```

Comment:
@borlesanket, wouldn't you want the uid and gid of the user that is submitting the job here?